python/lsst/ts/standardscripts/maintel/m1m3/check_actuators.py

```diff
@@ -0,0 +1,245 @@
+        metadata.duration = self.time_one_bump * (
+            len(self.actuators_to_test) + total_tested_secondary
+        )
+
+    def has_secondary_actuator(self, actuator_id: int) -> bool:
+        """Determines whether a given actuator has a
+        secondary axis or not.
+        """
+
+        return actuator_id in self.m1m3_secondary_actuator_ids
+
+    async def run(self):
+        await self.assert_feasibility()
+        start_time = time.time()
+
+        # Get M1M3 detailed state
+        detailed_state = DetailedState(
+            (
+                await self.mtcs.rem.mtm1m3.evt_detailedState.aget(
+                    timeout=self.mtcs.fast_timeout,
+                )
+            ).detailedState
+        )
+        self.log.info(f"Current M1M3 detailed state: {detailed_state!r}.")
+
+        # Put M1M3 engineering mode
+        await self.mtcs.enter_m1m3_engineering_mode()
+
+        for i, actuator_id in enumerate(self.actuators_to_test):
+            await self.checkpoint(f"Running bump test on FA ID: {actuator_id}.")
+
+            # Determine if actuator has a secondary actuator
+            secondary_exist = self.has_secondary_actuator(actuator_id)
+
+            await self.mtcs.run_m1m3_actuator_bump_test(
+                actuator_id=actuator_id,
+                primary=True,
+                secondary=secondary_exist,
+            )
+
+            # Getting test status
+            (
+                fa_id,
+                primary_status,
+                secondary_status,
+            ) = await self.mtcs.get_m1m3_bump_test_status()
+
+            if not secondary_exist:
+                self.log.info(
+                    f"Bump test on FA ID {fa_id}. Primary FA: "
+                    f"{primary_status.upper()}."
+                )
+
+            else:
+                self.log.info(
+                    f"Bump test on FA ID {fa_id}. Primary/Secondary: "
+                    f"{primary_status.upper()}/{secondary_status.upper()} ."
+                )
+
+            self.log.info(
+                f"Bump test done for {i + 1} of {len(self.actuators_to_test)}"
+            )
+
+        end_time = time.time()
+        elapsed_time = (end_time - start_time) / 60.0
+
+        # TO DO
+        # Add detailed information on each passed and, special, on each failed
+        await self.checkpoint(
+            f"M1M3 bump test completed. It took {elapsed_time:.2f} minutes"
+        )
+
+    async def cleanup(self):
+        if self.state.state != ScriptState.ENDING:
+            try:
+                self.log.warning("M1M3 bump test stopped. Killing actuator forces.")
+
+                await self.mtcs.stop_m1m3_bump_test()
+
+            except asyncio.TimeoutError:
+                self.log.exception(
+                    "Stopping bump test command timed out during cleanup procedure."
+                )
+            except Exception:
+                self.log.exception("Unexpected exception in stop_m1m3_bump_test.")
```

Comment:
Remember to exit engineering state at the end with:

```
await self.mtcs.exit_m1m3_engineering_mode()
```

Reply:
Is it ok giving a `log.info` message when entering/exiting engineering mode?

Reply:
The [mtcs](https://github.com/lsst-ts/ts_observatory_control/blob/d94417554a84c4384921042f82b38d23aac88967/python/lsst/ts/observatory/control/maintel/mtcs.py#L1033) already logs it...